subsys/bluetooth/controller/ll_sw/ull_df.c

```diff
@@ -665,4 +635,311 @@ static struct lll_df_adv_cfg *df_adv_cfg_acquire(void)
+		}
+		if (ext_hdr->aux_ptr) {
+			size_rem += sizeof(struct pdu_adv_aux_ptr);
+		}
+
+		if ((pdu->adv_ext_ind.ext_hdr_len - size_rem) != PDU_AC_EXT_HEADER_SIZE_MIN) {
+			return false;
+		}
+	}
+
+	return true;
+}
+#endif /* CONFIG_BT_CTLR_DF_PER_ADV_CTE_NUM_MAX && CONFIG_BT_CTLR_DF_PER_ADV_CTE_NUM_MAX > 1 */
+
+/*
+ * @brief Function removes content of cte_info field in periodic advertising chain.
+ *
+ * The function removes cte_info from extended advertising header in all PDUs in periodic
+ * advertising chain. If particular PDU is empty (holds cte_info only) it will be removed from
+ * chain.
+ *
+ * @param adv            Pointer to periodic advertising set.
+ * @param df_cfg         Pointer to direction finding configuration
+ * @param[out] ter_idx   Pointer used to return index of allocated or updated PDU.
+ *                       Index is required for scheduling the PDU for transmission in LLL.
+ * @param[out] first_pdu Pointer to return address of first PDU in a chain
+ *
+ * @return Zero in case of success, other value in case of failure.
+ */
+static uint8_t rem_cte_info_from_per_adv_chain(struct ll_adv_set *adv,
+					       struct lll_df_adv_cfg *df_cfg, uint8_t *ter_idx,
+					       struct pdu_adv **first_pdu)
+{
+	void *extra_data_prev, *extra_data;
+	struct pdu_adv *pdu_prev, *pdu;
+	struct lll_adv_sync *lll_sync;
+	uint8_t pdu_rem_field_flags;
+	int err;
+
+	lll_sync = adv->lll.sync;
+
+	/* NOTE: ULL_ADV_PDU_HDR_FIELD_CTE_INFO is just information that extra_data
+	 * should be removed in case of this call ull_adv_sync_pdu_alloc.
+	 * Other flags here do not change anything. It may be changed to use
+	 * true/false flag for extra_data allocation.
+	 */
+	err = ull_adv_sync_pdu_alloc(adv, ULL_ADV_PDU_EXTRA_DATA_ALLOC_NEVER, &pdu_prev, &pdu,
+				     &extra_data_prev, &extra_data, ter_idx);
+	if (err != BT_HCI_ERR_SUCCESS) {
+		return err;
+	}
+
+	if (extra_data) {
+		ull_adv_sync_extra_data_set_clear(extra_data_prev,
+						  extra_data, 0,
+						  ULL_ADV_PDU_HDR_FIELD_CTE_INFO,
+						  NULL);
+	}
+
+	*first_pdu = pdu;
+
+	pdu_rem_field_flags = ULL_ADV_PDU_HDR_FIELD_CTE_INFO;
+
+#if defined(CONFIG_BT_CTLR_DF_PER_ADV_CTE_NUM_MAX) && (CONFIG_BT_CTLR_DF_PER_ADV_CTE_NUM_MAX > 1)
+	struct pdu_adv *pdu_new, *pdu_chained;
+	bool new_chain;
+
+	/* It is possible that the function is called after e.g. advertising data were updated.
+	 * In such situation the function will run on already allocated chain. Do not allocate
+	 * new chain then. Reuse already allocated PDUs and allocate new ones only if the chain
+	 * was not updated yet.
+	 */
+	new_chain = (pdu_prev == pdu ? false : true);
+
+	/* Get next PDU in a chain. Alway use pdu_prev because it points to actual
+	 * former chain.
+	 */
+	pdu_chained = lll_adv_pdu_linked_next_get(pdu_prev);
+
+	/* Go through existing chain and remove CTE info. */
+	while (pdu_chained != NULL) {
+		if (pdu_ext_adv_is_empty_without_cte(pdu_chained) == true) {
```

Comment:
```suggestion
		if (pdu_ext_adv_is_empty_without_cte(pdu_chained)) {
```

Reply:
Code was refactored in a similar way as here: https://github.com/zephyrproject-rtos/zephyr/pull/36585/commits/1f6c24c1c6ed5efc3e7f7c8e653f8ca0f8cf3d08#r675458125